<launch>
  <arg name="scenario"/>
  <arg name="default_queue_size" default="10"/>
  <arg name="max_robot_speed" default="0.5"/>
  <arg name="robot_mode" default="0"/>
  <arg name="robot_radius" default="2"/>
  <arg name="enable_groups" default="true"/>
  <arg name="simulation_factor" default="1.0"/> <!-- Speed up -->
  <arg name="update_rate" default="25.0"/> <!-- Hz -->
  <arg name="spawn_period" default="5.0"/>
  <arg name="people_frame_ref" default="map"/>

  <arg name="kbd_teleop" default="false"/>
  <arg name="rqt_teleop" default="false"/>
  <arg name="include_teleop_ped" default="false"/>
  <arg name="ped_x" default="0.0"/>
  <arg name="ped_y" default="0.0"/>
  <arg name="ped_theta" default="0.0"/>

  <param name="/hri/selected_agent_id" value="1000"/>


  <!-- robot_mode == 0: CONTROLLED implies that the robot's velocity is set by some planning algorithm like DWA. Its very similar to the teleoperation mode, using tf in the mix is just a layer on top of this -->
  <!-- robot_mode == 1: TELEOPERATION , we map the output of a teleop node (like joystick) to drive the robot -->
  <!-- robot_mode == 2: SOCIAL mode means the robot's velocity is set by the social force just like the other agents -->
  <!-- main simulator node -->
  <node name="pedsim_simulator" pkg="pedsim_simulator" type="pedsim_simulator" output="screen">
    <param name="scene_file" value="$(find hrisim_pedsim)/scenarios/$(arg scenario).xml" type="string"/>
    <param name="default_queue_size" value="$(arg default_queue_size)" type="int"/>
    <param name="max_robot_speed" value="$(arg max_robot_speed)" type="double"/>
    <param name="robot_mode" value="$(arg robot_mode)" type="int"/>
    <param name="robot_radius" value="$(arg robot_radius)" type="double"/>
    <param name="enable_groups" value="$(arg enable_groups)" type="bool"/>
    <param name="simulation_factor" value="$(arg simulation_factor)" type="double"/>
    <param name="update_rate" value="$(arg update_rate)" type="double"/>
    <param name="spawn_period" value="$(arg spawn_period)" type="double"/>
    <param name="frame_id" value="$(arg people_frame_ref)" type="string"/>
  </node>

  <!-- Pedestrian simulated by differential robot -->
  <group ns="ped/control" if="$(arg include_teleop_ped)">
    <!-- pedestrian driving controller -->
    <node name="ped_controller" type="simulate_diff_drive_ped" pkg="pedsim_simulator" output="screen">
      <param name="ped_x" value="$(arg ped_x)"/>
      <param name="ped_y" value="$(arg ped_y)"/>
      <param name="ped_theta" value="$(arg ped_theta)"/>
      <param name="world_frame" value="$(arg people_frame_ref)"/>
    </node>

  </group>

  <!-- Remapping cmd_vel -->
  <node pkg="topic_tools" type="relay" name="my_relay" args="/mobile_base_controller/cmd_vel /pedbot/control/cmd_vel"/>

  <!-- Visualizer -->
  <include file="$(find pedsim_visualizer)/launch/visualizer.launch"/>

</launch>